### PR TITLE
New module: `metabat2`

### PR DESCRIPTION
**Pull request by maxibor**
Add metabat2 binning module (see #820 )

- [x] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [x] Emit the `versions.yml` file.
- [x] Follow the naming conventions.
- [x] Follow the parameters requirements.
- [x] Follow the input/output options guidelines.
- [x] Add a resource `label`
- [x] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
    - [x] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd`
    - [ ] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd`
    - [x] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd`

### CODE REVIEW

File: modules/metabat2/jgisummarizebamcontigdepths/main.nf [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    tuple val(meta), path(bam), path(bai)
+
+    output:
+    tuple val(meta), path("*.depth.txt.gz"), emit: depth
+    path "versions.yml"                    , emit: versions
+
+    script:
+    def prefix = options.suffix ? "${meta.id}${options.suffix}" : "${meta.id}"
+    """
+    OMP_NUM_THREADS=$task.cpus
```
[the remark this answers is not in the document]
Comment (maxibor) [reply modules/metabat2/jgisummarizebamcontigdepths/main.nf]: Thanks for spotting this. I look at the code from the MAG module, so there might the issue there as well

File: tests/modules/metabat2/metabat2/main.nf [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+        .map { it -> [it[0], it[1], []] }
+        .set { input_fasta }
+    input_fasta_alone
+        .join(METABAT2_JGISUMMARIZEBAMCONTIGDEPTHS.out.depth)
+        .set { input_fasta_depth }
+    
+    
+
+    METABAT2_NO_DEPTH ( input_fasta )
+    METABAT2_DEPTH ( input_fasta_depth )
```
[the remark this answers is not in the document]
Comment (maxibor) [reply tests/modules/metabat2/metabat2/main.nf]: They are now split into two different workflows

File: modules/metabat2/metabat2/meta.yml [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+      description: File containing software versions
+      pattern: "versions.yml"
+  - fasta:
+      type: file
+      description: Bins created from assembled contigs in fasta file
+      pattern: "*.fa.gz"
+  - membership:
+      type: file
+      description: cluster memberships as a matrix format.
+      pattern: "*.membership.tsv.gz"
```
Comment (jfy133) [inline modules/metabat2/metabat2/meta.yml:52]:
```suggestion
      pattern: "*.tsv.gz"
```

File: tests/modules/metabat2/metabat2/main.nf [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+
+    METABAT2_JGISUMMARIZEBAMCONTIGDEPTHS ( input_depth )
+
+    Channel.fromPath(params.test_data['sarscov2']['genome']['genome_fasta'], checkIfExists: true)
+        .map { it -> [[ id:'test', single_end:false ], it] }
+        .set { input_fasta_alone }
+
+    input_fasta_alone
+        .map { it -> [it[0], it[1], []] }
+        .set { input_fasta }
```
Comment (jfy133) [inline tests/modules/metabat2/metabat2/main.nf:22]: Shouldn't this be removed entirely if you don't have the depth file for this test?

File: tests/modules/metabat2/metabat2/test.yml [hunk 5]
```diff
@@ -0,0 +1,19 @@
+- name: metabat2 metabat2 test_metabat2_no_depth
+  command: nextflow run tests/modules/metabat2/metabat2 -entry test_metabat2_no_depth -c tests/config/nextflow.config
+  tags:
+    - metabat2
+    - metabat2/metabat2
+  files:
+    - path: output/metabat2/bins/test.1.fa.gz
+    - path: output/metabat2/test.depth.txt.gz
```
Comment (jfy133) [inline tests/modules/metabat2/metabat2/test.yml:8]:
```suggestion
    - path: output/metabat2/test.txt.gz
```

File: tests/modules/metabat2/metabat2/test.yml [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+    - path: output/metabat2/test.membership.tsv.gz
```
Comment (jfy133) [inline tests/modules/metabat2/metabat2/test.yml:9]:
```suggestion
    - path: output/metabat2/test.tsv.gz
```

File: tests/modules/metabat2/metabat2/test.yml (hunk 5 above)
Comment (jfy133) [reply tests/modules/metabat2/metabat2/test.yml:8]: If you want to add `depth`, you can update the `options` definitions in the `include` statment of the tests to include the prefix

File: tests/modules/metabat2/metabat2/test.yml (hunk 6 above)
Comment (jfy133) [reply tests/modules/metabat2/metabat2/test.yml:9]: If you want to add `membership`, you can update the `options` definitions in the `include` statment of the tests to include the prefix

**Review by jfy133 — APPROVED** [review]

**Pull request merged**